### PR TITLE
[Compute] `az disk-access`: Add new command group to operate disk access resource

### PR DESCRIPTION
**Pull request by houk-ms**
**Backgroud** 

Feature Request: CLI Support for private links of managed disks #13917 

This PR is a part of supporting private links for managed disks. Private links for managed disks is a little different from the already supported private links in CLI (e.g, keyvault). It use the disk-access as an intermediate resource rather than connect with the target resource directly. Therefore, to support the private links for managed disks, we have to do the following things. This PR works to support the first step. 

> **1. Support CURD of the new resource disk-access**
> 2. Support create/update a disk/snapshot to associate it a disk-access resource
> 3. Support private links for managed disks

**Description**  

Disk access is a new resource designed for private links of managed disks, which can be associated with a managed disk or a snapshot as an intermediate resource to export the target disk via private network. 

- This PR adds a new command group `az disk-access` to operate the disk access resource. 
- This PR registers the disk-access operation group to compute in profile.

```
Group
    az disk-access : Manage disk access resources.

Commands:
    create : Create a disk access resource.
    delete : Delete a disk access resource.
    list   : List disk access resources.
    show   : Get information of a disk access resource.
    update : Update a disk access resource.
    wait   : Place the CLI in a waiting state until a condition of a disk access is met.
```
```
Command
    az disk-access create : Create a disk access resource.

Arguments
    --name -n           [Required] : Name of the disk access resource.
    --resource-group -g [Required] : Name of resource group. You can configure the default group
                                     using `az configure --defaults group=<name>`.
    --location -l                  : Location. Values from: `az account list-locations`. You can
                                     configure the default location using `az configure --defaults
                                     location=<location>`.
    --no-wait                      : Do not wait for the long-running operation to finish.
    --tags                         : Space-separated tags: key[=value] [key[=value] ...]. Use "" to
                                     clear existing tags.

Examples
    Create a disk access resource.
        az disk-access create -g MyResourceGroup -l centraluseuap -n MyDiskAccess
```
```
Command
    az disk-access update : Update a disk access resource.

Arguments
    --no-wait           : Do not wait for the long-running operation to finish.
    --tags              : Space-separated tags: key[=value] [key[=value] ...]. Use "" to clear
                          existing tags.

Generic Update Arguments
    --add               : Add an object to a list of objects by specifying a path and key value
                          pairs.  Example: --add property.listProperty <key=value, string or JSON
                          string>.
    --force-string      : When using 'set' or 'add', preserve string literals instead of attempting
                          to convert to JSON.
    --remove            : Remove a property or an element from a list.  Example: --remove
                          property.list <indexToRemove> OR --remove propertyToRemove.
    --set               : Update an object by specifying a property path and value to set.  Example:
                          --set property1.property2=<value>.

Resource Id Arguments
    --ids               : One or more resource IDs (space-delimited). It should be a complete
                          resource ID containing all information of 'Resource Id' arguments. You
                          should provide either --ids or other 'Resource Id' arguments.
    --name -n           : Name of the disk access resource.
    --resource-group -g : Name of resource group. You can configure the default group using `az
                          configure --defaults group=<name>`.
    --subscription      : Name or ID of subscription. You can configure the default subscription
                          using `az account set -s NAME_OR_ID`.

Examples
    Update a disk access resource.
        az disk-access update -g MyResourceGroup -n MyDiskAccess --tags tag1=val1 tag2=val2
```

**Testing Guide**

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

File: src/azure-cli/azure/cli/command_modules/vm/commands.py [hunk 1]
```diff
@@ -210,6 +216,13 @@ def load_command_table(self, _):
         g.show_command('show', 'get')
         g.custom_command('list', 'list_disk_encryption_sets')
 
+    with self.command_group('disk-access', compute_disk_access_sdk, operation_group='disk_accesses', client_factory=cf_disk_accesses, min_api='2020-05-01') as g:
+        g.custom_command('create', 'create_disk_access', supports_no_wait=True)
+        g.generic_update_command('update', custom_func_name='update_disk_access', supports_no_wait=True)
+        g.show_command('show', 'get')
+        g.custom_command('list', 'list_disk_accesses')
+        g.command('delete', 'delete')
```
Comment (arrownj) [inline src/azure-cli/azure/cli/command_modules/vm/commands.py:224]: need to add a `wait` command.

File: src/azure-cli/azure/cli/command_modules/vm/custom.py [hunk 2]
```diff
@@ -3362,3 +3362,25 @@ def update_disk_encryption_set(instance, client, resource_group_name, key_url=No
     return instance
 
 # endregion
+
+
+# region Disk Access
+def create_disk_access(cmd, resource_group_name, disk_access_name, location, tags=None, no_wait=False):
```
Comment (arrownj) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py:3368]: Can `location` be alternative ? We usually use resource group default location for a specific resource if location is not specified.

Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py:3368]: You can put `client` in argument list

Comment (houk-ms) [reply src/azure-cli/azure/cli/command_modules/vm/custom.py]: updated.

File: src/azure-cli/azure/cli/command_modules/vm/commands.py (hunk 1 above)
Comment (houk-ms) [reply src/azure-cli/azure/cli/command_modules/vm/commands.py:225]: sure

File: src/azure-cli/azure/cli/command_modules/vm/custom.py [hunk 3]
```diff
@@ -3362,3 +3362,31 @@ def update_disk_encryption_set(instance, client, resource_group_name, key_url=No
...
+def list_disk_accesses(cmd, client, resource_group_name=None):
+    if resource_group_name:
+        return client.list_by_resource_group(resource_group_name)
+    return client.list()
+
+
+def update_disk_access(instance, tags=None):
+    if tags is not None:
+        instance.tags = tags
+    return instance
```
Comment (fengzhou-msft) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py:3383]: This method is not used?

Comment (houk-ms) [reply src/azure-cli/azure/cli/command_modules/vm/custom.py:3383]: not used, will be removed

**Pull request merged**